### ISSUE TITLE
forced subtitle mkv flag are not preserved

### ISSUE THREAD

**Issue by Mangoes** [donmelton/video_transcoding#76]
Hi,

**The Issue:**

I want to preserve the mkv forced subtitle flags, but transcode-video removes them.

_(I absolutely do not want to burn in subtitles, as all my mkvs have the forced subtitles flag properly set. Kodi automatically switches on the correct forced subtitles depending on the audio language at playback, exactly like a DVD/Bluray player. So burning them in is much less efficient as it doesn't support forced languages on the fly switching)._

I primarily use transcode-video to only reencode the video and copy over all the audio and subtitles tracks.

How can I make it preserve all the mkv flags ? (fyi Megui correctly preserves the mkv forced subtitle flags when opening a source).

This is my command line:
`transcode-video --crop detect --copy-audio all --add-subtitle all --no-auto-burn --quick "c:\movie.mkv`

I have also tried to add `--force-subtitle scan` to this command line, but it doesn't change anything (the forced subtitle flag is still removed)

I see no `--copy-subtitle all` option in the `--help`. Would a `--copy-subtitle all` preserve the forced flags?

**Question:**

Does `--force-subtitle scan` do the same thing as makemkv?

As often the subtitles are not marked as forced inside the PGS, **the optimal solution would be:**

1. always preserved the original mkv forced flag (this is what `--add-subtitle all` should do) and
2. also scan the PGS subtitle for forced lines (so when running a batch we also automatically flag the subtitles in case the original had no forced flag but was containing a PGS with forced lines), this is what `--force-subtitle scan` does (?).

**To summarize:**

1. Could you please fix `--copy-subtitle` to preserve the original mkv forced flags
2. and if we add `--force-subtitle scan` do we have the best "catch all forced subtiles solution" ?

Many thanks.

**Comment by Mangoes** [comment]
And I forgot, when running an auto-batch, so that we don't find ourselves with 1 subtitles muxed several times in the mkv with the above "catch all forced subtiles solution":

3 - remove any double subtitle tracks (in case one subbtiltes apprears 2 times: once flagged as forced and once not flagged, this is what makemkv sometimes does)

Thanks a lot.

**Comment by donmelton** [comment]
@Mangoes Thanks for the writing up this issue. I'll take a look at in a few days when I back at my Mac.

**Comment by donmelton** [comment]
@Mangoes Sorry I took so long to look at your issue. I've been traveling a lot this summer and this is the first week I've been able to really focus on the project.

There is no `-copy-subtitle` option. It's `--add-subtitle` and I've been very clear in the past that its behavior is _not_ to set attribute flags and that users must explicitly choose which subtitle to burn or force. That's why `--burn-subtitle` and `--force-subtitle` options exist. It's also why the `scan` argument is available for those latter two options.

So, I don't intend to change the semantics or behavior of `--add-subtitle`. Doing so might also cause problems for other users since there's no way to "un-force" a subtitle.

Anyway, using `--force-subtitle scan` with `--add-subtitle all` should get you the behavior you want. I just tried it myself and it found the subtitle with the forced flag in the input, added it to the output, and then correctly set the forced flag in the output. As a bonus, there's no need to use the `no-auto-burn` option when using that combination. Automatic subtitle burning is disabled in that case.

Regarding the `scan` implementation, that's internal to HandBrake itself. I don't know how it works or how similar it is to MakeMKV. In fact, it's a bit of a pain in the ass for me since I can't tell what HandBrake has decided to do until the output file is complete. I then have to scan the output file looking for a subtitle with the default flag and set the forced flag in the "cleanup" portion of my code.

So, can you try `--force-subtitle scan --add-subtitle all` and see if that works for you? If it doesn't, I will consider adding something like a `--auto-force` option to do this.

Thanks.

**Comment by donmelton** [comment]
@Mangoes Any update on trying `--force-subtitle scan --add-subtitle all` to see what happens?

Unless I hear from you soon I'll close this issue. Of course, you can still comment even if it's closed, and we can discuss re-opening it.

Thanks.

**Comment by donmelton** [comment]
@Mangoes Since I haven't heard from you, I'm going to assume this is resolved and close the issue now. But, like I said, you can still comment and we can discuss re-opening it.
